hecat/processors/url_check.py

```diff
@@ -42,13 +44,23 @@ def check_return_code(url, current_item_index, total_item_count, errors):
             return True
         else:
             error_msg = '{} : HTTP {}'.format(url, response.status_code)
-            logging.error('[%s/%s] %s', current_item_index, total_item_count, error_msg)
-            errors.append(error_msg)
+            logging.warning('[%s/%s] %s', current_item_index, total_item_count, error_msg)
+            handle_failed_url(url, failed_urls_file, escalation_limit)
             return False
+    # Invalid URL should be handled as an error directly
+    except (requests.exceptions.InvalidURL, requests.exceptions.InvalidSchema) as invalid_url_error:
+        error_msg = '{} : {}'.format(url, invalid_url_error)
+        logging.error('[%s/%s] %s', current_item_index, total_item_count, error_msg)
+        handle_failed_url(url, failed_urls_file, escalation_limit)
```

Comment:
```suggestion
        logging.error('[%s/%s] %s', current_item_index, total_item_count, error_msg)
        errors.append(error_msg)
```

There is no need to wait for the escalation limit to be reached here, an invalid URL will always stay invalid, it should always trigger an error and make the script exit with an error code at the end (which happens if `errors` list is not empty)